### DISCUSSION TITLE
Suggestion: How to Farm

### DISCUSSION THREAD

**Discussion by andbruu** [voxeltycoon/issues#362] [General]
I can see that Farms are currently part of the Roadmap, but lacking a description (as far as I can tell). I would like to suggest doing farms in a way that is a bit different from other "resource nodes", to spice up the game a bit.

First off, farms should not depend on resource nodes, and should be placeable anywhere. To not make the placement of farms trivial however, farms would be more effective if placed in a relative flat, open area, perhaps with a boost from nearby (fresh) water.

The player will place the main farm building. Unlike other resource buildings, this building will NOT produce any resources. It will instead place a field a few times a month (production cycle). When placing a field, it will choose from a variety of shapes of between 20-30 squares, and place it within one height difference of the farm (the demand for flat areas). It will also place the fields so there is at least two tiles clearing to any other entity (road, rail, building, field, ect.).

Each field will then act as a producer, but with a relative low production rate and capacity. Nearby stations will be able to pull from fields, as they would pull from mines.

If a farm is about to place a field, but cannot find space for it, it will instead boost the production rate of all its fields, until it next will try to place a field (at which point it will continue the boost, if there is still no space for a field.

To take it a step further, tractors could be introduced, able to drive off-road and even through fields. They would be to slow to be of much use outside of farm work. A tractors "station" could be a silo, that unlike other stations do not come with a bit of road or track. Place a few siloes at the most distant fields, and one near a warehouse and rail-station, and you have increased the number of fields that the rail station is able to collect from.

Finally, have the output generated by the fields be a generic "crop", and have the visuals change dependent on the biome of the field. Some areas may have corn fields, while others will have wheat or veggie. Its all just visual, and the output of farms and fields are just "Crop". Works bother for processing into flour, feed for animals (with ranches) and direct sale (farmers' markets).

**Comment by victorleblais** [comment]
Good initiative ! Feedback (even for planned features) is always welcome.
